includes/course-theme/class-sensei-course-theme-templates.php

```diff
@@ -112,6 +119,128 @@ public function should_use_quiz_template() {
 		return false;
 	}
 
+	/**
+	 * Returns a list of Learning Mode templates that are available.
+	 *
+	 * @throws Error If the extra templates provided by third party does not abide by the required structure.
+	 */
+	public static function get_available_block_templates() {
+		$base_path = Sensei_Course_Theme::instance()->get_course_theme_root() . '/templates';
```

Comment:
I know it fits well in the `_Templates` class, but that file is getting a bit large. Maybe they could live in something like `Sensei_Course_Theme_Template_Selection`?

Reply:
Moved to `Sensei_Course_Theme_Temeplate_Selection` class.

Also created the `Sensei_Course_Theme_Template` class that represents the individual templates.

Not sure about the names though. Let me know if you want to rename those classes to something else.